inst/pages/98_exercises.qmd

```diff
@@ -1158,62 +1155,145 @@ See the OMA book for examples.
+# Load necessary libraries
+library(mia)
+library(SummarizedExperiment)
+library(vegan)
+library(bluster)
+library(scater)
+
+# Generate a sample SummarizedExperiment object
+data(GlobalPatterns)
+tse <- GlobalPatterns
 
-# Perform k-means clustering on the reduced dimensions of pca_aitchison
-pca_aitchison_clusters <- kmeans(reducedDim(pca_aitchison), centers = 3)$cluster
+# Perform MDS on the abundance data using scater::runMDS
+# Specify the correct assay to be used
+tse <- runMDS(tse, FUN = vegan::vegdist, method = "bray", assay.type = "counts")
 
-# Plot the reduced dimensions of pca_aitchison with clusters as colors
-plotReducedDim(pca_aitchison, colour_by = pca_aitchison_clusters)
+# Perform k-means clustering (k = 3) using the bluster package
+set.seed(123)
+clustering_result <- clusterRows(reducedDim(tse, "MDS"), BLUSPARAM=KmeansParam(centers=3))
+colData(tse)$cluster <- as.factor(clustering_result)
+
+# Visualize MDS with clusters
+plotReducedDim(tse, dimred = "MDS", colour_by = "cluster") +
+  labs(title = "MDS with k-means Clustering") +
+  theme_minimal() +
+  scale_color_manual(values = c("red", "blue", "green"))
 ```
-Where : 
-'reducedDim' is a function that extracts the reduced dimensionality representation of your data (e.g., from a PCA or 
-PCoA object).
-'plotReducedDim' is a function that can plot these reduced dimensions and has a 'colour_by argument' that accepts a vector of cluster assignments.
 
-#### 2. Heatmap with Annotations:
+### 2: Cluster Analysis with Heatmap Visualization
 
-* Annotate the heatmap with additional metadata (e.g., subject groups or sample types).
+- Hierarchical clustering of samples based on abundance data.
+- Visualizes the clustered data using a heatmap, with cluster information added to metadata.
 
 ```{r}
+#| label: heatmap-visualization-ex2
+#| eval: FALSE
+#| code-fold: true
+#| code-summary: "Show solution"
+
 # Load necessary libraries
+library(mia)
 library(SummarizedExperiment)
-library(pheatmap)
+library(vegan)
+library(bluster)
+library(scater)
+library(ComplexHeatmap)
+library(circlize)
+library(compositions)
+
+# Generate a sample SummarizedExperiment object 
+data(GlobalPatterns)
+tse <- GlobalPatterns
 
-# Example data creation
-# Assuming enterotype is a SummarizedExperiment object with sample metadata
-# and assay_data is a matrix containing assay data
+# Extract the abundance data matrix and perform CLR transformation
+clr_data <- transformAssay(se, method = "clr", assay = "counts")
+
+# Perform hierarchical clustering on the samples using Euclidean distance
+distance_matrix <- dist(clr_data, method = "euclidean")
+clustering <- hclust(distance_matrix, method = "average")
+
+# Cut the dendrogram to form clusters (k = 3 for this example)
+clusters <- cutree(clustering, k = 3)
+
+# Add cluster information to colData
+colData(se)$cluster <- as.factor(clusters)
+
+# Visualize the heatmap with clusters using ComplexHeatmap
+Heatmap(clr_data, 
+        name = "CLR Counts",
```

Comment:
Use schem package since you are plotting abundance data

Reply:
I rewrote the codes for exercise 2 and 3 using sechm package, could you review it once when possible?

Reply:
Seems ok